src/Infer/Solve.hs

```diff
@@ -738,7 +788,7 @@ exportedExps Slv.AST { Slv.aexps, Slv.apath } = case apath of
   bundleExports _ exp = return $ case exp of
     e'@(Slv.Solved _ _ (Slv.Export (Slv.Solved _ _ (Slv.Assignment n _)))) ->
       (n, e')
-    (Slv.Solved _ _ (Slv.TypedExp e'@(Slv.Solved _ _ (Slv.Export (Slv.Solved _ _ (Slv.Assignment n _)))) _))
+    e'@(Slv.Solved _ _ (Slv.TypedExp (Slv.Solved _ _ (Slv.Export (Slv.Solved _ _ (Slv.Assignment n _)))) _))
```

Comment:
What's happening here exactly? Seems like a lot has been packed into one line.

Reply:
I deconstruct the exp and make sure that it has that shape. So an Export can be one of the following:
- Export Assignment
- TypedExp Export Assignment

That `e'` I bind is the Exp from which I take the type in order to populate the ENV of the consumer. I was wrongly picking the one of the Export and not the TypedExp which led to some surprising errors. Because sometimes the resolution of say a function could be something like a -> String, but the Typing says it's a Bool -> String. Then what we want in the env is obviously Bool -> String and not a -> String. Typing in some cases makes the type of expressions more precise.